### PR TITLE
fix a couple bugs, simplify getMatches

### PR DESCRIPTION
**Pull request by fgregg**
- Simplifies get matches
- Fixes bug where we weren't handling low confidence matches that had got filtered out in populatehumanreview
- Mark records for review, as we review them. should have much better estimates of remaining count.

### CODE REVIEW

File: api/utils/delayed_tasks.py [hunk 1]
```diff
@@ -367,6 +359,26 @@ def populateHumanReview(session_id):
     worker_session.add(dedupe_session)
     worker_session.commit()
 
+
+def markAsReviewed(session_id, record_id, reviewer) :
+    dedupe_session = worker_session.query(DedupeSession).get(session_id)
```
Comment (evz) [inline api/utils/delayed_tasks.py]: Minor thing but we don't actually need this dedupe_session here do we?

Comment (fgregg) [inline api/utils/delayed_tasks.py]: That's right. I'll take it out.

File: api/utils/helpers.py [hunk 2]
```diff
@@ -343,22 +342,32 @@ def getMatches(session_id, records):
...
+                best_records[entity] = dict(record)
+                best_records[entity]['confidence'] = float(confidence)
+
+        matches.append((messy_records[messy_id], 
+                        list(best_records.values())))
+            
     del deduper
-    return all_matches, unmatched_records
+
+    return matches
```
Comment (evz) [inline api/utils/helpers.py]: What happens to the records that were not matched? Meaning, if we get 100 records out of the database and find only 40 that have possible matches, what happens to the other 60?

Comment (fgregg) [inline api/utils/helpers.py]: The action is on line 355. If no matches were found for messy_id in possible_matches, then return an empty list of possible matches.

Comment (evz) [inline api/utils/helpers.py]: Cool. Merge imminent.

**Pull request merged**